rust/kernel/buffer.rs

```diff
@@ -0,0 +1,43 @@
+// SPDX-License-Identifier: GPL-2.0
+
+//! Struct for writing to a pre-allocated buffer with the [`write!`] macro.
+//!
+//! [`write!`]: https://doc.rust-lang.org/core/macro.write.html
```

Comment:
```suggestion
//! Struct for writing to a pre-allocated buffer with the [`write!`] macro.
```

Intra-doc links (for `core` and `alloc`, not `std` since we don't use it) should work automatically (or did you try and it was not working?).

Reply:
Ah, I didn't try it. Was just following the pattern for existing links.

Looks like items in `core` and `alloc` link to `doc.rust-lang.org` even without explicitly setting them. We're not generating our own docs for  `core` and `alloc` - at least I don't see any rules for them in the Makefile and they don't seem to be generated by `make rustdoc`.

Reply:
Yeah, they go to the official docs at the moment, but perhaps we should generate them too. I can look into that. Let me create an issue.

Reply:
#97